### ISSUE TITLE
Missing retention policy tables in the Cloud Sandbox

### ISSUE THREAD

**Issue by DanielGoehler** [microsoft/navcontainerhelper#2810]
**Describe the issue**
Compared to the OnPrem sandbox, why does the cloud sandbox only have the 8888 Email Outbox and 8889 Sand Mail tables for setting up a retention policy?

**Scripts used to create container and cause the issue**
```
New-NavContainer `
-accept_eula `
-containername test-cloud `
-artifactUrl "https://bcartifacts.azureedge.net/sandbox/21.1.48363.50227/de" `
-PublicDnsName docker.company.com `
-useSSL `
-useTraefik `
-shortcuts None `
-auth NavUserPassword `
-Credential $credential `
-licenseFile "C:\ProgramData\company\dockerLicenseFiles\company.bclicense"
```

**Full output of scripts**
```
BcContainerHelper version 3.0.10
BcContainerHelper emits usage statistics telemetry to Microsoft
BcContainerHelper is version 3.0.10
BcContainerHelper is running as administrator
Hyper-V is Enabled
UsePsSession is True
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 19.03.5
Docker Server Version is 19.03.5
Removing Session test-cloud
Removing container test-cloud
Removing entries from hosts
Removing test-cloud from container hosts file
Removing test-cloud-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\test-cloud
Fetching all docker images
Fetching all docker volumes
Enabling SSL as otherwise all clients will see mixed HTTP / HTTPS request, which will cause problems e.g. on the mobile and modern windows clients
Using image mcr.microsoft.com/businesscentral:10.0.17763.3650
PublicDnsName is docker.company.com
Creating Container test-cloud
Style: sandbox
Multitenant: Yes
Version: 21.1.48363.50227
Platform: 21.0.48357.50182
Generic Tag: 1.0.2.13
Container OS Version: 10.0.17763.3650 (ltsc2019)
Host OS Version: 10.0.17763.3650 (ltsc2019)
Using process isolation
Using locale de-DE
Adding special CheckHealth.ps1 to enable Traefik support
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\ProgramData\company\dockerLicenseFiles\company.bclicense
Additional Parameters:
-e webserverinstance=test-cloud
-e publicdnsname=docker.company.com
-l "traefik.protocol=https"
-l "traefik.web.frontend.rule=PathPrefix:/test-cloud"
-l "traefik.web.port=443"
-l "traefik.soap.frontend.rule=PathPrefix:/test-cloudsoap;ReplacePathRegex: ^/test-cloudsoap(.*) /BC$1"
-l "traefik.soap.port=7047"
-l "traefik.rest.frontend.rule=PathPrefix:/test-cloudrest;ReplacePathRegex: ^/test-cloudrest(.*) /BC$1"
-l "traefik.rest.port=7048"
-l "traefik.dev.frontend.rule=PathPrefix:/test-clouddev;ReplacePathRegex: ^/test-clouddev(.*) /BC$1"
-l "traefik.dev.port=7049"
-l "traefik.snap.frontend.rule=PathPrefix:/test-cloudsnap;ReplacePathRegex: ^/test-cloudsnap(.*) /BC$1"
-l "traefik.snap.port=7083"
-l "traefik.dl.frontend.rule=PathPrefixStrip:/test-clouddl"
-l "traefik.dl.port=8080"
-l "traefik.dl.protocol=http"
-l "traefik.enable=true"
-l "traefik.frontend.entryPoints=https"
--env customNavSettings=PublicODataBaseUrl=https://docker.company.com/test-cloudrest/odata,PublicSOAPBaseUrl=https://docker.company.com/test-cloudsoap/ws,PublicW
ebBaseUrl=https://docker.company.com/test-cloud
Files in C:\ProgramData\BcContainerHelper\Extensions\test-cloud\my:
- AdditionalOutput.ps1
- CheckHealth.ps1
- license.bclicense
- MainLoop.ps1
- SetupVariables.ps1
- updatecontainerhosts.ps1
Creating container test-cloud from image mcr.microsoft.com/businesscentral:10.0.17763.3650
461a2455d317d220a6bfb92457087a653f61860cb108fc509757a33b9d613514
Waiting for container test-cloud to be ready
Using artifactUrl https://bcartifacts.azureedge.net/sandbox/21.1.48363.50227/de
Using installer from C:\Run\210-new
Installing Business Central
Installing from artifacts
Starting Local SQL Server
Starting Internet Information Server
Copying Service Tier Files
c:\dl\sandbox\21.1.48363.50227\platform\ServiceTier\Program Files
c:\dl\sandbox\21.1.48363.50227\platform\ServiceTier\System64Folder
Copying PowerShell Scripts
c:\dl\sandbox\21.1.48363.50227\platform\WindowsPowerShellScripts\Cloud\NAVAdministration
c:\dl\sandbox\21.1.48363.50227\platform\WindowsPowerShellScripts\WebSearch
Copying Web Client Files
c:\dl\sandbox\21.1.48363.50227\platform\WebClient\Microsoft Dynamics NAV
Copying ModernDev Files
c:\dl\sandbox\21.1.48363.50227\platform
c:\dl\sandbox\21.1.48363.50227\platform\ModernDev\program files\Microsoft Dynamics NAV
Copying additional files
Copying ConfigurationPackages
C:\dl\sandbox\21.1.48363.50227\de\ConfigurationPackages
Copying Test Assemblies
C:\dl\sandbox\21.1.48363.50227\platform\Test Assemblies
Copying Extensions
C:\dl\sandbox\21.1.48363.50227\de\Extensions
Copying Applications
C:\dl\sandbox\21.1.48363.50227\platform\Applications
Copying Applications.DE
C:\dl\sandbox\21.1.48363.50227\de\Applications.DE
Copying dependencies
Copying ReportBuilder
Importing PowerShell Modules
Restoring CRONUS Demo Database
Setting CompatibilityLevel for tenant on localhost\SQLEXPRESS
Exporting Application to CRONUS
Removing Application from tenant
Modifying Business Central Service Tier Config File for Docker
Creating Business Central Service Tier
Installing SIP crypto provider: 'C:\Windows\System32\NavSip.dll'
Starting Business Central Service Tier
Importing license file
Copying Database on localhost\SQLEXPRESS from tenant to default
Taking database tenant offline
Copying database files
Attaching files as new Database default
Putting database tenant back online
Mounting tenant database
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Stopping Business Central Service Tier
Installation took 257 seconds
Installation complete
Initializing...
Setting host.containerhelper.internal to 172.20.96.1 in container hosts file
Starting Container
Hostname is test-cloud
PublicDnsName is docker.company.com
Using NavUserPassword Authentication
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 7E79C1A5F88374FF19C8998C2F2F8BF3BD236566
DNS identity docker.company.com
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting PublicODataBaseUrl to https://docker.company.com/test-cloudrest/odata
Setting PublicSOAPBaseUrl to https://docker.company.com/test-cloudsoap/ws
Setting PublicWebBaseUrl to https://docker.company.com/test-cloud
Starting Service Tier
CertificateThumprint 7E79C1A5F88374FF19C8998C2F2F8BF3BD236566
Registering event sources
Creating DotNetCore Web Server Instance
Using application pool name: test-cloud
Using default container name: NavWebApplicationContainer
Copy files to WWW root C:\inetpub\wwwroot\test-cloud
Create the application pool test-cloud
Create website: NavWebApplicationContainer with SSL
Update configuration: navsettings.json
Done Configuring Web Client
Enabling Financials User Experience
Using license file 'c:\run\my\license.bclicense'
Import License
Dismounting Tenant
Mounting Tenant
Mounting Database for default on server localhost\SQLEXPRESS with AllowAppDatabaseWrite = False
Sync'ing Tenant
Tenant is Operational
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
Creating SUPER user
Container IP Address: 172.20.109.23
Container Hostname  : test-cloud
Container Dns Name  : docker.company.com
Web Client          : https://docker.company.com/test-cloud/?tenant=default
Dev. Server         : https://docker.company.com
Dev. ServerInstance : BC
Dev. Server Tenant  : default
Setting test-cloud-default to 127.0.0.1 in container hosts file

Files:
http://docker.company.com:8080/ALLanguage.vsix
http://docker.company.com:8080/certificate.cer

Container Total Physical Memory is 511.9Gb
Container Free Physical Memory is 204.1Gb

Initialization took 45 seconds
Ready for connections!
Reading CustomSettings.config from test-cloud
Container test-cloud successfully created
Because of Traefik, the following URLs need to be used when accessing the container from outside your Docker host:
Web Client:        https://docker.company.com/test-cloud
SOAP WebServices:  https://docker.company.com/test-cloudsoap
OData WebServices: https://docker.company.com/test-cloudrest
Dev Service:       https://docker.company.com/test-clouddev
Snapshot Service:  https://docker.company.com/test-cloudsnap
File downloads:    https://docker.company.com/test-clouddl

Use:
Get-BcContainerEventLog -containerName test-cloud to retrieve a snapshot of the event log from the container
Get-BcContainerDebugInfo -containerName test-cloud to get debug information about the container
Enter-BcContainer -containerName test-cloud to open a PowerShell prompt inside the container
Remove-BcContainer -containerName test-cloud to remove the container again
docker logs test-cloud to retrieve information about URL's again
...
```

**Screenshots**
![Screenshot](https://user-images.githubusercontent.com/9556549/205074048-7814de6a-bf43-4afc-9c81-3a75505c0f33.png)

**Comment by freddydk** [comment]
I have no idea, sorry. I don't think this is related to docker, but more to our demo data.
Please post the question in Yammer (you can link to this issue if you like)

**Comment by DanielGoehler** [comment]
Sure, I will do this. 

Just to make sure, did the BaseApp gets published in the creation process of the docker? 

(Uninstall and Install of the BaseApp fixes this issues.)

**Comment by freddydk** [comment]
the databases are part of the artifacts generated - no apps are unpublished during container generation

**Comment by DanielGoehler** [comment]
Done: https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=2029199931154432

**Comment by grobyns** [comment]
Hi all,
the problem seems to be that we don't run the Company-Initialize codeunit and that is where we (also) initialize the allowed tables for retention policies. 
I've repro'ed it like this: In SaaS, create a new company and pick production / setup data only. Wait for the whole thing to finish, and then log in. When you log in we check whether we need to run Company-Initialize. This check just verifies whether a General Ledger Setup record exists. Here it does, because we imported setup data using rapid start when we created the company. Company-Initialize doesn't run and so the retention policy tables aren't added.
If you do the same thing but pick no data when you create the company, then the tables will be added on first log in. 

We're working on this issue so it should be fixed soon.
